x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/NativeMemoryCapacity.java

```diff
@@ -55,32 +55,49 @@ NativeMemoryCapacity merge(NativeMemoryCapacity nativeMemoryCapacity) {
     public AutoscalingCapacity autoscalingCapacity(int maxMemoryPercent, boolean useAuto) {
         // We calculate the JVM size here first to ensure it stays the same given the rest of the calculations
         final Long jvmSize = useAuto
-            ? Optional.ofNullable(this.jvmSize).orElse(dynamicallyCalculateJvmSizeFromNativeMemorySize(node))
+            ? Optional.ofNullable(this.jvmSize).orElse(dynamicallyCalculateJvmSizeFromNativeMemorySize(nodeMlNativeMemoryRequirement))
             : null;
-        // We first need to calculate the actual node size given the current native memory size.
+        // We first need to calculate the required node size given the required native ML memory size.
         // This way we can accurately determine the required node size AND what the overall memory percentage will be
-        long actualNodeSize = NativeMemoryCalculator.calculateApproxNecessaryNodeSize(node, jvmSize, maxMemoryPercent, useAuto);
+        long requiredNodeSize = NativeMemoryCalculator.calculateApproxNecessaryNodeSize(
+            nodeMlNativeMemoryRequirement,
+            jvmSize,
+            maxMemoryPercent,
+            useAuto
+        );
         // We make the assumption that the JVM size is the same across the entire tier
         // This simplifies calculating the tier as it means that each node in the tier
         // will have the same dynamic memory calculation. And thus the tier is simply the sum of the memory necessary
         // times that scaling factor.
-        double memoryPercentForMl = NativeMemoryCalculator.modelMemoryPercent(actualNodeSize, jvmSize, maxMemoryPercent, useAuto);
+        //
+        // Since this is a _minimum_ node size, the memory percent calculated here is not
+        // necessarily what "auto" will imply after scaling. Because the JVM occupies a
+        // smaller proportion of memory the bigger the node, the memory percent might be
+        // higher than we calculate here, potentially resulting in a bigger ML tier than
+        // required. The effect is most pronounced when the minimum node size is small and
+        // the minimum tier size is a lot bigger. For example, if the minimum node size is
+        // 1GB and the total native ML memory requirement for the tier is 32GB then the memory
+        // percent will be 41%, implying a minimum ML tier size of 78GB. But in reality a
+        // single 64GB ML node would have an auto memory percent of 90%, and 90% of 64GB is
+        // plenty big enough for the 32GB of ML native memory required.
+        // TODO: improve this in the next refactoring
```

Comment:
Fixing this isn't what this PR is primarily about, but I noticed it while debugging the new test, so thought it worth documenting.  The way all our sliding scales interact is very complex.  We should refactor this once the formula for node size to JVM size is reversible.

To be more accurate this method also needs to know the maximum ML node size and the principle that Cloud always scales up before scaling out.  Maximum ML node size is not currently reliably set.  But work is in progress to make it accurate, and this will take effect no later than the delegation of the JVM heap size to Elasticsearch.

So in the next refactor all the prerequisites will be in place to make this better.